MdePkg/Test/Mock/Library/GoogleTest/MockUefiBootServicesTableLib/MockUefiBootServicesTableLib.cpp

```diff
@@ -50,54 +59,89 @@ extern "C" {
+  gBS_ReinstallProtocolInterface,                                                      // EFI_REINSTALL_PROTOCOL_INTERFACE
+  gBS_UninstallProtocolInterface,                                                      // EFI_UNINSTALL_PROTOCOL_INTERFACE
+  gBS_HandleProtocol,                                                                  // EFI_HANDLE_PROTOCOL
+  NULL,                                                                                // VOID
+  NULL,                                                                                // EFI_REGISTER_PROTOCOL_NOTIFY
+  NULL,                                                                                // EFI_LOCATE_HANDLE
+  gBS_LocateDevicePath,                                                                // EFI_LOCATE_DEVICE_PATH
+  NULL,                                                                                // EFI_INSTALL_CONFIGURATION_TABLE
+  NULL,                                                                                // EFI_IMAGE_LOAD
+  NULL,                                                                                // EFI_IMAGE_START
+  NULL,                                                                                // EFI_EXIT
+  NULL,                                                                                // EFI_IMAGE_UNLOAD
+  NULL,                                                                                // EFI_EXIT_BOOT_SERVICES
+  NULL,                                                                                // EFI_GET_NEXT_MONOTONIC_COUNT
+  NULL,                                                                                // EFI_STALL
+  NULL,                                                                                // EFI_SET_WATCHDOG_TIMER
+  NULL,                                                                                // EFI_CONNECT_CONTROLLER
+  NULL,                                                                                // EFI_DISCONNECT_CONTROLLER
+  gBS_OpenProtocol,                                                                    // EFI_OPEN_PROTOCOL
+  gBS_CloseProtocol,                                                                   // EFI_CLOSE_PROTOCOL
+  NULL,                                                                                // EFI_OPEN_PROTOCOL_INFORMATION
+  NULL,                                                                                // EFI_PROTOCOLS_PER_HANDLE
+  gBS_LocateHandleBuffer,                                                              // EFI_LOCATE_HANDLE_BUFFER
+  gBS_LocateProtocol,                                                                  // EFI_LOCATE_PROTOCOL
+  gBS_InstallMultipleProtocolInterfaces,                                               // EFI_INSTALL_MULTIPLE_PROTOCOL_INTERFACES
+  (EFI_UNINSTALL_MULTIPLE_PROTOCOL_INTERFACES)gBS_UninstallMultipleProtocolInterfaces, // EFI_UNINSTALL_MULTIPLE_PROTOCOL_INTERFACES

```

Comment:
Is the cast needed here?

Reply:
Yes, there was a compilation error without this cast, please see error below:

ERROR - Compiler #2440 from c:\psns\MU_BASECORE\MdePkg\Test\Mock\Library\GoogleTest\MockUefiBootServicesTableLib\MockUefiBootServicesTableLib.cpp(140): 'initializing': cannot convert from 'EFI_STATUS (__cdecl *)(EFI_HANDLE *,...)' to 'EFI_UNINSTALL_MULTIPLE_PROTOCOL_INTERFACES'